src/main/scala/com/microsoft/hyperspace/index/sources/iceberg/IcebergRelationMetadata.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.sources.iceberg
+
+import com.microsoft.hyperspace.index.Relation
+import com.microsoft.hyperspace.index.sources.FileBasedRelationMetadata
+
+class IcebergRelationMetadata(metadata: Relation) extends FileBasedRelationMetadata {
+
+  override def refresh(): Relation = {
+    metadata.copy(options = metadata.options - "snapshot-id" - "as-of-timestamp")
+  }
+
+  override def internalFileFormatName(): String = {
+    "parquet"
+  }
+
+  override def enrichIndexProperties(properties: Map[String, String]): Map[String, String] = {
```

Comment:
nit; could you add comments for above functions?

Reply:
Why should we duplicate comments while Javadoc does it for us? I know there is a place for duplication, but I don't understand what's the benefit here. On the other hand, the disadvantage seems obvious: those comments can go out of sync pretty quickly.

Reply:
There can be some additional comments & reference for the developer.
And also for consistency.

Reply:
If the overridden method needs a description about the specific implementation, then it is not duplicating the information and it's fine by me.

I don't think the second one (reference) is a good enough reason to warrant duplication and risking out of sync content, because if a developer wants to know about the interface, then he/she should look up the comments in the interface itself. Even if he/she finds those in the implementation source code, he/she cannot be sure it is the same/up to date with the interface (because it is often not the case), and should do another look up anyway.

Therefore, I suggest removing all Scaladoc comments from overriding methods if they are the same as those in the overridden methods, for consistency.

Reply:
seems fine to remove comments if they are duplicates, but this creates inconsistencies with the existing code as @sezruby brings up, and we can handle this as a separate PR.